### PR TITLE
Update RBAC to allow creating events

### PR DESCRIPTION
**Pull request by primeroz**
the controller is trying to create events for, at least, leadership election

```
capi-garbage-collector-5986579885-8fgkz capi-garbage-collector E0123 17:39:54.786410       1 event.go:267] Server rejected event '&v1.Event{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{
Name:"ba6eg525.giantswarm.io.173d00df2110f322", GenerateName:"", Namespace:"giantswarm", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UT
C), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ZZZ_D
eprecatedClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, InvolvedObject:v1.ObjectReference{Kind:"Lease", Namespace:"giantswarm", Name:"ba6eg525.giantswarm.io", UID:"168337ea-182e-481e-9972-fa07ae838
c74", APIVersion:"coordination.k8s.io/v1", ResourceVersion:"10187309", FieldPath:""}, Reason:"LeaderElection", Message:"capi-garbage-collector-5986579885-8fgkz_700bfcb4-3c96-4e6b-b1b9-44910eac11bc became leader"
, Source:v1.EventSource{Component:"capi-garbage-collector-5986579885-8fgkz_700bfcb4-3c96-4e6b-b1b9-44910eac11bc", Host:""}, FirstTimestamp:time.Date(2023, time.January, 23, 17, 39, 54, 784158498, time.Local), La
stTimestamp:time.Date(2023, time.January, 23, 17, 39, 54, 784158498, time.Local), Count:1, Type:"Normal", EventTime:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Series:(*v1.EventSeries)(nil), Action:"", 
Related:(*v1.ObjectReference)(nil), ReportingController:"", ReportingInstance:""}': 'events is forbidden: User "system:serviceaccount:giantswarm:capi-garbage-collector" cannot create resource "events" in API gro
up "" in the namespace "giantswarm"' (will not retry!)  
```

This PR:

- adds rbac for the capi-garbage-collector SA to allow creating events

### Checklist

- [ ] Update changelog in CHANGELOG.md.
- [ ] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

**Review by AverageMarcus — APPROVED** [review]

File: helm/capi-garbage-collector/templates/rbac.yaml [hunk 1]
```diff
@@ -31,6 +31,17 @@ rules:
       - get
       - create
       - update
+  - apiGroups:
+    - ""
+    resources:
+    - events
+    verbs:
+    - create
+    - get
```
Comment (fiunchinho) [inline helm/capi-garbage-collector/templates/rbac.yaml]: Does it need all the other verbs apart from `create`?

Comment (primeroz) [reply helm/capi-garbage-collector/templates/rbac.yaml]: Good point, i just copied the block from the `azure cloud controller` yeah will leave only create :+1:

also need to see about this failing check for a vulnerability

**Comment by primeroz** [comment]
@AverageMarcus @fiunchinho Actually i _think_ the fix for this issue identified by `nancy` _sonatype-2022-6522_ is not in any release yet

```
 λ git branch --contains 76a233ebe
* master

➜ git tag --contains 76a233ebe   

```

should i revert my `go.mod` changes and set it to ignore ?

**Comment by fiunchinho** [comment]
If there is no release with a fix just yet and you don't need to update, then I wouldn't update yet. But it's your call.

**Comment by primeroz** [comment]
tested and working

**Pull request merged**